comictaggerlib/issueidentifier.py

```diff
@@ -459,104 +460,151 @@ def search(self) -> list[IssueResult]:
         # now sort the list by name length
         series_second_round_list.sort(key=lambda x: len(x["name"]), reverse=False)
 
-        # build a list of volume IDs
-        volume_id_list = []
-        for series in series_second_round_list:
-            volume_id_list.append(series["id"])
+        # If the sources lacks issue level data, don't look for it.
+        if not self.sources_api.sources[self.sources_api.source].source_details["static_options"]["has_issues"]:
```

Comment:
This needs to end up shorter if you want to change the instance variable to talker that might make more sense
this:
`self.sources_api.sources[self.sources_api.source]`

Needs to become this:
`self.source` or `self.talker`

Most of the code should not be making any decisions on what talker it will be using it should just be given the object to use. The only places that it should decide this are on initial startup (when it first loads the settings) and in the settings window where it's theoretically possible to change it.

Reply:
I've created the active talker as sources_api.talker as it seems clearer and doesn't require making another var. Is that good enough or should I do:
```
self.sources_api = sources_api
self.talker = sources_api.talker
```
Is there something to be gained I don't understand? Not trying to be a PITA :)

Reply:
First off I have not gone through your code properly to see how everything works, so far I've mainly commented on individual issues that I've seen.
There are two particular issues that I have with this line regardless of any other details:
1. it is a minimum of 60 characters to retrieve a setting, that is way too long if 60 is the minimum you are probably doing something wrong.
   This being the minimum line that is 60 characters:
   `s.sources[t].source_details["static_options"]["has_issues"]`
1. If i already have the object I'm trying to use `self.sources_api` I'm not going to go figure it out again `self.sources_api.sources[self.sources_api.source]` because what's the purpose in using `self.sources_api` then?

Now to answer your questions.
Mainly keeping the API small and defined and not depending on implementation details, there should only be a specific set of information passed between SourcesAPI and the rest of ComicTagger and it should for the most part all be done through one class without going into attached classes (eg. sources_api.talker).
Looking at the `SourcesAPI` class, this class is your API. Every time that ComicTagger interacts with with a source, right now it's only ComicVine but it could be anything else, it should only touch the `SourcesAPI` class it shouldn't be touching whatever implementation is used, in this case `ComicVineTalker` because next time it might be something else and it may not support whatever it is that was attempted.
So what you need is a way to set and retrieve settings on the `SourcesAPI` class and specifically these settings need to be source-agnostic, your `static_options` is at the very least a place to start. That doesn't mean limiting the options to the absolute minimum it just means that they always need to be set to something even if the source doesn't use them.
Right now the minimal use of `SourcesAPI` with your latest changes is:
```python
# get the object
api = SourcesAPI()
# check an internal implementation detail
issues = api.talker.source_details["static_options"]["has_issues"]
# use api
if issues:
  print(api.fetch_issue_data_by_issue_id(1))
else:
  print(api.fetch_volume_data(1))
```

It should be:
```python
# get the object
api = SourcesAPI()
# check an if the api supports issues
issues = api.has_issues
# or
issues = api.settings.has_issues
# use api
if issues:
  print(api.fetch_issue_data_by_issue_id(1))
else:
  print(api.fetch_volume_data(1))
```
In the example I used a property but using a function is standard as well `api.settings().has_issues`.
I would probably use a custom class or a named tuple for the settings on the `SourcesAPI` class. A TypedDict is useful but it's primary purpose is for providing type hints when you normally can't such as when reading json from an web API.
The same argument goes for `SourceDetails` I would probably merge static_options and `SourceDetails` into one object to put all the info in the same place but that is not necessary. 

Don't worry about it, if anything I'm the PITA.

Reply:
Hopefully the commit to objectify source_details and static_option is there or there abouts. I was a little confused by the class def using TypedDict which didn't create objects. I assumed it would. Finding (or my bad searching) info on those sorts of things seems difficult.

When it comes to what to expect from a source, in the module loading code I've put checks in for the required fields.
```
required_fields_details = ["name", "id"]
required_fields_static = ["has_issues", "has_alt_covers", "has_censored_covers"]
required_fields_settings = ["enabled", "url_root"]
```
Because it's only returning the static known CV talker atm, I've skipped them.

Reply:
Yeah, python sometimes doesn't have documentation that explains things very well.
This is the actual specification of the TypedDict https://peps.python.org/pep-0589/
This is a good explanation of how it works from mypy https://mypy.readthedocs.io/en/stable/more_types.html#typeddict

Putting `static_options` on to the `SourceAPI` works